apollo-router/src/plugins/cache/entity.rs

```diff
@@ -518,6 +511,29 @@ impl EntityCache {
             supergraph_schema,
         })
     }
+
+    // Returns boolean to know if cache is enabled for this subgraph
+    fn subgraph_enabled(&self, subgraph_name: &str) -> bool {
+        self.enabled
+            // If subgraphs.all is set but there's no override for the subgraph
+            && ((self.subgraphs.all.enabled.unwrap_or(true) // Because enabled is true by default
+                && matches!(self.subgraphs.get(subgraph_name).enabled, Some(true) | None))
+                // It overrides subgraphs.all
+                || (self.subgraphs.all.enabled == Some(false) && matches!(self.subgraphs.get(subgraph_name).enabled, Some(true))))
```

Comment:
Sorry but I find it hard to follow the logic here. What do you think of reorganizing this into a `match`? I _think_ this is equivalent:

```rust
if !self.enabled {
    return false
}
match (
    self.subgraphs.all.enabled,
    self.subgraphs.get(subgraph_name).enabled,
) {
    (_, Some(x)) => x, // explicit per-subgraph setting overrides the `all` default
    (Some(true) | None, None) => true, // unset defaults to true
    (Some(false), None) => false,
}

Reply:
I agree with you, I planned to come back to this once the tests were good but I forgot